bsmetadata/experiments/with_metadata.py

```diff
@@ -40,88 +40,113 @@ def get_dataloaders(tokenizer, args):
             )
         if extension == "jsonl":
             extension = "json"
-        raw_datasets = load_dataset(extension, data_files=data_files, cache_dir=args.cache_dir)
+        datasets = load_dataset(extension, data_files=data_files, cache_dir=args.cache_dir)
 
-        if "validation" not in raw_datasets.keys():
-            raw_datasets["validation"] = load_dataset(
+        if "validation" not in datasets.keys():
+            datasets["validation"] = load_dataset(
                 extension,
                 data_files=data_files,
                 split=f"train[:{args.validation_split_percentage}%]",
                 cache_dir=args.cache_dir,
             )
-            raw_datasets["train"] = load_dataset(
+            datasets["train"] = load_dataset(
                 extension,
                 data_files=data_files,
                 split=f"train[{args.validation_split_percentage}%:]",
                 cache_dir=args.cache_dir,
             )
+    logger.info("Dataset loaded")
     # See more about loading any type of standard or custom dataset (from files, python dict, pandas DataFrame, etc) at
     # https://huggingface.co/docs/datasets/loading_datasets.html.
 
     # Preprocessing the datasets.
-    column_names = raw_datasets["train"].column_names
+    column_names = datasets["train"].column_names
+
+    logger.info("Start to add metadata and chunk examples")
 
     # First we pre-process our text and metadata
-    lm_datasets = raw_datasets.map(
-        functools.partial(add_metadata_and_chunk_examples, tokenizer=tokenizer, cfg=args),
+    datasets = datasets.map(
+        functools.partial(add_metadata_and_chunk_examples, tokenizer=tokenizer, cfg=args.metadata_config),
         batched=True,
         num_proc=args.preprocessing_num_workers,
         load_from_cache_file=not args.overwrite_cache,
         desc="Pre-process the text and metadata to create new samples",
         remove_columns=column_names,
+        batch_size=args.map_batch_size,
     )
+    logger.info("Add metadata and chunk examples finished")
 
     def create_labels_column(examples):
         examples["labels"] = examples["input_ids"].copy()
         return examples
 
+    logger.info("Create labels column")
     # Then we add the column containing the labels
-    lm_datasets = lm_datasets.map(
+    datasets = datasets.map(
         create_labels_column,
         batched=True,
         num_proc=args.preprocessing_num_workers,
         load_from_cache_file=not args.overwrite_cache,
         desc="Create labels column",
+        batch_size=args.map_batch_size,
     )
+    logger.info("Creating labels column finished")
+
+    train_dataset = datasets["train"]
+    val_dataset = datasets["validation"]
 
-    train_dataset = lm_datasets["train"]
-    val_dataset = lm_datasets["validation"]
+    logger.info(f"  Num train examples = {len(train_dataset)}")
+    logger.info(f"  Num validation examples = {len(val_dataset)}")
```

Comment:
I love this change and just wonder if any of you also like to see the numbers before tokenization and block-grouping/chunking.

Reply:
That is a good idea! :slightly_smiling_face: I'd like this addition too